pkg/controller/grafanadashboard/dashboard_controller.go

```diff
@@ -180,6 +184,11 @@ func (r *ReconcileGrafanaDashboard) importDashboard(d *i8ly.GrafanaDashboard) (r
 		return reconcile.Result{}, defaultErrors.New("operator namespace not yet known")
 	}
 
+	json, err := r.loadDashboardFromURL(d)
```

Comment:
This means we're downloading the JSON in every reconciliation (which happens every 15 seconds: https://github.com/integr8ly/grafana-operator/blob/master/pkg/controller/grafanadashboard/dashboard_controller.go#L128). What do you think about the following approach:

1. modify the `hasDashboardChanged` function to include the URL in the generated hash: https://github.com/integr8ly/grafana-operator/blob/master/pkg/controller/grafanadashboard/dashboard_controller.go#L136
2. Download the JSON only when a change is detected (there would always be a change on the first import). A change here means the URL or the json field are updated.

This of course has the problem that we don't automatically pull changes from remote dashboards. But i think that's  ok. We could at some point provide a webhook that forces the operator to refresh the dashboards from the URLs.

Reply:
makes sense